From: SeongJae Park <sj@kernel.org>
Date: Sun, 17 Nov 2024 13:02:02 -0800
Subject: [PATCH] mm/damon/core: fix uninitialized 'ret' usage from
 kdamond_callback()

kdamond_callback() can return uninitialized value when the callback is
canceled.  Initialize the return value always.

Fixes: 26e20409cfae7 ("mm/damon/core: simplify damon_call() callback path")
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202411180218.mUqglaqC-lkp@intel.com/
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 88bd2dc40a26..1a0bf9ad139a 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2016,7 +2016,7 @@ static void kdamond_usleep(unsigned long usecs)
 static int kdamond_callback(struct damon_ctx *ctx, bool cancel)
 {
 	struct damon_call_req *req;
-	int ret;
+	int ret = 0;
 
 	mutex_lock(&ctx->call_request_lock);
 	req = ctx->call_request;
-- 
2.39.5

